cmd/cloud/server.go

```diff
@@ -85,12 +89,16 @@ var serverCmd = &cobra.Command{
 			"working-directory": wd,
 		}).Info("Starting Mattermost Provisioning Server")
 
+		privateRoute53ZoneID, _ := command.Flags().GetString("private-route53-id")
+		privateDNS, _ := command.Flags().GetString("private-dns")
 		// Setup the provisioner for actually effecting changes to clusters.
 		kopsProvisioner := provisioner.NewKopsProvisioner(
 			clusterRootDir,
 			s3StateStore,
 			certificateSslARN,
 			logger,
+			aws.New(privateRoute53ZoneID),
```

Comment:
This is my primary concern in this PR. I can see how storing an aws client in the provisioner would be useful, but this is a bit of an anti-pattern.

If you look here, you will see how we are passing in an aws client to the Installation supervisor for use with the installation changes. This should be mimicked for `NewClusterSupervisor` instead of this implementation.

Also, we should probably modify the aws client itself to hold a DNS value so we can store the private DNS in the client itself.

Reply:
As discussed, I added a comment that aws is used only for Route53 registration and todo to move it to clustersupervisor in the future.